repository/src/raw/tests.rs

```diff
@@ -448,3 +450,25 @@ async fn remote() {
         )]
     );
 }
+
+#[tokio::test]
+async fn reserved_state() {
+    let td = TempDir::new().unwrap();
+    let path = td.path();
+    let mut repo = init_repository_with_initial_commit(path).await.unwrap();
+
+    let (rs, _) = simperby_test_suite::generate_standard_genesis(10);
+
+    repo.checkout(MAIN.into()).await.unwrap();
+    repo.create_semantic_commit(SemanticCommit {
+        title: "test".to_owned(),
+        body: "test-body".to_owned(),
+        diff: Diff::Reserved(Box::new(rs.clone())),
+    })
+    .await
+    .unwrap();
+    repo.checkout(MAIN.into()).await.unwrap();
```

Comment:
After making a commit, the branch is automatically changed to point at the new commit. 
So this line might not be needed, too.
But please check after removing this.

Reply:
ok. fixed